src/App.js

```diff
@@ -1,48 +1,93 @@
 import React, { Component } from 'react';
-import { movieData } from './apiCalls.js';
+import { movieData, movieDetailsById } from './apiCalls.js';
 import Movies from './Movies.js';
 import './App.css';
+import {Route, Switch, Link} from 'react-router-dom';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
       movies: [],
+      selectedMovie: {},
       isSinglePoster: false,
     }
   }
 
-  componentDidMount() {
-    movieData()
+  componentDidMount = async () => {
+    await movieData()
     .then(movies => this.setState({ movies: movies.movies }))
     .catch(error => console.log(error))
   }
 
-  findPoster = (id) => {
-    const selectedMovie = this.state.movies.filter(movie => movie.id === id);
-
-    this.setState({ movies: selectedMovie, isSinglePoster: true });
+  singleMovieFetch = async (id) => {
+    const movieDetails = await movieDetailsById(id);
+    this.setState({
+      selectedMovie: movieDetails,
+    })
   }
 
   setMovieInfo = () => {
+    const currentMovie = this.state.selectedMovie.movie;
     return(
-      <section className='movie-info'>
-        <p>{this.state.movies[0].title}</p>
-        <p>Average Rating: {this.state.movies[0].average_rating.toFixed(1)}</p>
-        <p>Release Date: {this.state.movies[0].release_date}</p>
-        <button className='go-back-btn' onClick={() => window.location.reload()}>Go Back</button>
-      </section>
+        <section className='movie-info'>
+          {currentMovie !== undefined && <p>{currentMovie.title}</p>}
+          {currentMovie !== undefined && this.state.selectedMovie.movie.tagline.length > 0 && <p>{currentMovie.tagline}</p>}
+          {currentMovie !== undefined && <p>Average Rating: {currentMovie.average_rating.toFixed(1)}</p>}
+          {currentMovie !== undefined && <p>Release Date: {currentMovie.release_date}</p>}
+          {currentMovie !== undefined && <p>Overview: {currentMovie.overview}</p>}
+          {currentMovie !== undefined && <p>Genre(s): {currentMovie.genres.map(genre => `${genre}, `)}</p>}
+          {currentMovie !== undefined && <p>Runtime: {currentMovie.runtime} minutes</p>}
```

Comment:
This can probably be refactored.  I think we could do some conditional rendering for the whole section.  Let's loom into that.